core/mempool/src/lib.rs

```diff
@@ -26,15 +29,16 @@ pub struct Pool {
     // transactions need to be grouped by account, and within each account,
     // ordered by nonce so that runtime will not ignore some of the transactions
     transactions: RwLock<HashMap<AccountId, BTreeMap<u64, SignedTransaction>>>,
+    // store the map that allows us to lookup transaction by hash
+    transaction_info: RwLock<HashMap<CryptoHash, (AccountId, u64)>>,
     receipts: RwLock<HashSet<ReceiptBlock>>,
     storage: Arc<RwLock<ShardChainStorage>>,
     trie: Arc<Trie>,
     state_viewer: TrieViewer,
-    snapshots: RwLock<HashSet<ChainPayload>>,
-    /// List of requested snapshots that can't be fetched yet.
-    pending_snapshots: RwLock<Vec<(AuthorityId, CryptoHash)>>,
     /// List of requested snapshots that are unblocked and can be confirmed.
-    ready_snapshots: RwLock<Vec<(AuthorityId, CryptoHash)>>,
+    snapshots: RwLock<HashSet<Snapshot>>,
+    /// List of requested snapshots that can't be fetched yet.
+    pending_snapshots: RwLock<HashSet<Snapshot>>,
```

Comment:
What is the logic behind having a `Snapshot` instead of its hash in `pending_snapshots`? To my understanding we either have an entire snapshot or we have none of its transactions.

Reply:
The snapshot itself has to be stored somewhere.